File: java/WebFormsCore/ExtensionWebFormsMethods.java
package WebFormsCore;

// WebForms.java 2.0 - The Back-End Part of WebForms Core Technology, Owned by Elanat (https://elanat.net)
// Compatible with WebFormsJS version 2.0

public class ExtensionWebFormsMethods {
    /// <summary>
    /// This Method Does Not Support QueryAll
    /// </summary>
    public static String appendPlace(String text, String value) {
        if (text == null || text.length() < 1)
            return value;

        return text + "|" + value;
    }

    public static String appendParent(String text) {
        return "/" + text;
    }

    public static String exportActionControlsToWebFormsTag(String actionControls, boolean addLine) {
        return (addLine ? "\n" : "") + "<web-forms ac=\"" + actionControls + "\"></web-forms>";
    }

    public static String exportActionControlsToWebFormsTag(String actionControls) {
        return exportActionControlsToWebFormsTag(actionControls, false);
    }

    public static String exportActionControlsToHtmlComment(String actionControls, boolean addLine) {
        return (addLine ? "\n" : "") + "<!--[web-forms]\n" + actionControls + "-->";
    }

    public static String exportActionControlsToHtmlComment(String actionControls) {
        return exportActionControlsToHtmlComment(actionControls, false);
    }

    public static String exportActionControlsToResponse(String actionControls) {
        return "[web-forms]\n" + actionControls;
    }

    public static String removeOuter(String text, String startString, String endString) {
        int start = text.indexOf(startString);
        if (start == -1)
            return text;

        int end = text.indexOf(endString, start);
        if (end == -1)
            return text;

        int lengthToRemove = (end - start) + endString.length();

        return text.substring(0, start) + text.substring(start + lengthToRemove);
    }

    public static String lineBreak(String text) {
        return text.replace("\n", "$[sln]");
    }
}

File: elixir/webforms_core/extension_web_forms_methods.ex
# webforms.ex 2.1 - The Back-End Part of WebForms Core Technology, Owned by Elanat (https://elanat.net)
# Compatible with WebFormsJS version 2.1

defmodule WebFormsCore.ExtensionWebFormsMethods do
  def child(text, value) do
    if text == "" do
      value
    else
      text <> "|" <> value
    end
  end

  def parent(text) do
    cond do
      text == "" -> text
      String.ends_with?(text, "|/") or String.ends_with?(text, "//") -> text <> "/"
      true -> text <> "|/"
    end
  end

  def criteria(text, value) do
    if text == "" do
      value
    else
      text <> "?" <> (value |> String.replace("|", "$[vb];") |> String.replace("?", "$[qu];"))
    end
  end

  def append_fetch_replace(text, search_value, value) do
    fs = <<28>>
    text = String.slice(text, 1..-1)
    "@;" <> search_value <> fs <> value <> fs <> text
  end

  def line_break(text, encode_line \\ false) do
    encode = if encode_line, do: "$[sln];", else: ""

    text
    |> String.replace("\r\n", encode)
    |> String.replace("\n", encode)
    |> String.replace("\r", encode)
  end

  # Converts Numbers to Strings
  def to_js_string(text), do: "\"" <> text <> "\""

  # Get JS Object Momentary
  def to_js_object(text), do: "$" <> text

  # Get JS Object Returned Value Once
  def to_js_return_object(text), do: "$@" <> text
end